app/controllers/hostgroups_controller.rb

```diff
@@ -77,9 +76,10 @@ def destroy
 
   def current_parameters
     Taxonomy.as_taxonomy @organization, @location do
-      inherited_parameters = Hostgroup.find(params['hostgroup_parent_id']).inherited_params(true) if params['hostgroup_parent_id']
+      hostgroup_parent_parameters = Hostgroup.authorized(:view_hostgroups).find(params['hostgroup_parent_id']).parameters(true) if params['hostgroup_parent_id']
```

Comment:
`params['hostgroup_parent_id']` will be an empty string when no parent is set, so this always evaluates to true and so `Hostgroup..find('')` fails, generating a 404.